app/services/common_platform/api/hearing_fetcher.rb

```diff
@@ -6,7 +6,7 @@ class HearingFetcher < ApplicationService
       URL = "hearing/results"
 
       def initialize(hearing_id:, sitting_day:, connection: CommonPlatform::Connection.call)
-        @params = { hearingId: hearing_id, sittingDay: sitting_day }.compact
+        @params = { hearingId: hearing_id, sittingDay: sitting_day }.reject { |_k, v| v.blank? }
```

Comment:
For my learning, can you explain the difference here?

Reply:
`compact` rejects key/value pairs with a `nil` value only.

Rejecting all key/value pairs with a blank value means we also filter out empty strings.

```
irb> hash = {foo: "bar", baz: ""}

irb> hash.compact
=> {:foo=>"bar", :baz=>""}

irb)> hash.reject { |_k, v| v.blank? }
=> {:foo=>"bar"}
```

Reply:
Awesome, thank you!